pkg/nfd-master/node-updater-pool.go

```diff
@@ -0,0 +1,86 @@
+package nfdmaster
+
+import (
+	"sync"
+
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type nodeUpdaterPool struct {
+	queue     workqueue.RateLimitingInterface
+	wg        sync.WaitGroup
+	nfdMaster *nfdMaster
+	started   bool
+	mutex     sync.Mutex
+}
```

Comment:
One suggestion to potentially beautify a bit

```suggestion
type nodeUpdaterPool struct {
	sync.Mutex
	workqueue.RateLimitingInterface
	
	wg        sync.WaitGroup
	nfdMaster *nfdMaster
	started   bool
}
```

With this you could do simpler `u.Lock()` instead of `u.mutex.Lock()` and e.g. `pool.Add(nodeName)` instead of `pool.queue.NodeName()`. WDYT?

Reply:
```go
	u.queue.ShutDown()
	u.wg.Wait()
```
Would be transformed to:
```go
	u.ShutDown()
	u.wg.Wait()
```
Isn't it misleading to have `u.Shutdown()` and then `u.wg.Wait()`?

Reply:
It gives the intention that we're shutting down the pool and then waiting for the waitgroup, where in reality we're shutting down only the queue.

Reply:
Yeah, maybe it's cleaner to have queue as a separate field